gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -1306,6 +1307,83 @@ gum_do_enumerate_exports (GumEnumerateExportsContext * ctx,
   return carry_on;
 }
 
+kern_return_t
+gum_malloc_memory_reader (task_t remote_task,
+    vm_address_t remote_address, vm_size_t size, void ** local_memory) {
+
+  *local_memory = (void *) remote_address;
+
+  return KERN_SUCCESS;
+}
+
+struct _gum_malloc_helper_info {
+  GumFoundMallocRangeFunc func;
+  gpointer user_data;
+  gboolean should_abort;
+};
+
+void
+gum_process_enumerate_malloc_ranges_helper (task_t task,
+    void * user_data, unsigned type,
+    vm_range_t * ranges, unsigned count) {
+
+  struct _gum_malloc_helper_info * info =
+    (struct _gum_malloc_helper_info *) user_data;
+
+  if (info->should_abort) {
+    return;
+  }
+
+  GumMemoryRange gum_range;
+  GumMallocRangeDetails details;
+  details.range = &gum_range;
```

Comment:
Style-wise we tend to have an empty line after the declarations to make the code a bit more readable.